piffle/iiif.py

```diff
@@ -108,3 +108,125 @@ def format(self, image_format):
         img = self.get_copy()
         img.image_options['format'] = image_format
         return img
+
+    def parse_from_url(self, url):
+        '''Parses Image API parameters from URL provided
+        Per http://iiif.io/api/image/2.0/#image-request-uri-syntax, using slashes to navigate URL'''
+        
+        url_components = url.split('/')
+        
+        _quality, _format = url_components[-1].split('.')
+        _rotation = url_components[-2]
+        _size = url_components[-3]
+        _region = url_components[-4]
+        _image_id = url_components[-5]
+        _api_endpoint = '/'.join(url_components[:-6])
+        
+        # reinit
+        self.__init__(api_endpoint=_api_endpoint, image_id=_image_id, region=_region,
+                     size=_size, rotation=_rotation, quality=_quality, format=_format)
+                     
+    def api_params_as_dict(self):
```

Comment:
Is there a clearer/shorter name we could use? dict_opts ? 

Also, I think it might be helpful to add a little more to the description to elaborate on why this is useful (i.e., to allow logical decision making based on the image features).